### DISCUSSION TITLE
$event->addMeetLink() not working

### DISCUSSION THREAD

**Discussion by TSY-0730** [spatie/laravel-google-calendar#275] [Q&A]
I followed the instructions in the README.md to add a Google Meet link to an event using the `$event->addMeetLink()` method, but I'm encountering an error:

{ "error": { "errors": [ { "domain": "global", "reason": "invalid", "message": "Invalid conference type value." } ], "code": 400, "message": "Invalid conference type value." } }

Can anyone help me? Thank you.

**Comment by cardei** [comment]
Hi, Same for me, any solution ??

**Reply by jsterj** [reply]
Anyone ever solve this?  Having the same issue.

**Reply by noumansarwarr** [reply]
Please let me know if you are able to solve this I'm getting same here. @jsterj

**Reply by jsterj** [reply]
@noumansarwarr I never did get it resolved. I decided to try Zoom instead. I've been looking at these two Laravel-Zoom packages:

https://github.com/JubaerHossain/zoom-laravel

https://github.com/MacsiDigital/laravel-zoom

**Reply by YutaInouePH** [reply]
@noumansarwarr 

I encountered the same problem. You'll need to go to google suite admin and modify some security settings.

https://jimicious.medium.com/how-to-setup-googlemeet-in-rails-using-service-accounts-f1762ec645de
**Giving access**

After activating the Domain-Wide delegation setting to the service account, I was able to generate the google meet link. 
<img width="600" alt="Screenshot 0007-01-28 at 15 58 35" src="https://github.com/user-attachments/assets/5c002e8c-e6c0-41ff-9a42-37332c709bcd" />

Make sure to set the `GOOGLE_CALENDAR_IMPERSONATE` environment variable. 

At this moment, I cannot find another way to make this work except for this step.

**Reply by jsterj** [reply]
Thanks for the info. I ended up going with Zoom, but this will be helpful if I try Google again.

**Comment by syntx1997** [comment]
Hello, I just find the solution to the issue: don't forget to put env variable: GOOGLE_CALENDAR_IMPERSONATE
